Patch: 11_handle_propfind.patch
Author: Tollef Fog Heen <tfheen@debian.org>
Submitted upstream: http://sourceforge.net/tracker/index.php?func=detail&aid=1578756&group_id=103&atid=300103
Don't fall flat on our face if we get a request type which we
don't understand (Closes: #202821)
Index: mailman-2.1.12/scripts/driver
===================================================================
--- mailman-2.1.12.orig/scripts/driver	2009-03-02 20:41:55.000000000 +0100
+++ mailman-2.1.12/scripts/driver	2009-03-02 20:50:00.000000000 +0100
@@ -98,6 +98,15 @@
         module = getattr(pkg, scriptname)
         main = getattr(module, 'main')
         try:
+            import os
+            request_method = os.environ.get('REQUEST_METHOD')
+            if not request_method in ['GET', 'POST', 'HEAD']:
+                print "Status: 405 Method not allowed"
+                print "Content-type: text/plain"
+                print
+                print "The method is not allowed"
+                sys.exit()
+                
             try:
                 sys.stderr = logger
                 sys.stdout = tempstdout
